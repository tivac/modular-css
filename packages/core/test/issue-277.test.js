"use strict";

var plugin = require("../plugins/scoping.js"),

    processor = require("postcss")([ plugin ]);

function process(src, options) {
    return processor.process(
        src,
        Object.assign(Object.create(null), {
            from  : "packages/core/test/specimens/a.css",
            namer : (file, selector) => `a_${selector}`
        },
        options || {})
    );
}

describe("/issues", () => {
    describe("/277", () => {
        it("should transform multiple :global(...) in a single selector", () => {
            expect(
                process(":global(.a) :global(.b) { color: red; }").css
            )
            .toMatchSnapshot();
        });
    });
});
