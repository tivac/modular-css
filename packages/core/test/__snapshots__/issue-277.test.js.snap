// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/issues /277 should transform multiple :global(...) in a single selector 1`] = `".a .b { color: red; }"`;
